Deliverance/DeliveranceMailChimpList.php

```diff
@@ -1081,749 +559,118 @@ public function getInterests()
-				'Campaign “Subject” is null');
-		}
-
-		$from_address = $campaign->getFromAddress();
-		if ($from_address == null) {
-			throw new DeliveranceCampaignException(
-				'Campaign “From Address” is null');
-		}
-
-		$from_name = $campaign->getFromName();
-		if ($from_name == null) {
-			throw new DeliveranceCampaignException(
-				'Campaign “From Name” is null');
-		}
-
-		$to_name = $campaign->getToName();
-
-		$options = array(
-			'list_id'      => $this->shortname,
-			'title'        => $title,
-			'subject'      => $subject,
-			'from_email'   => $from_address,
-			'from_name'    => $from_name,
-			'to_name'      => $to_name,
-			'authenticate' => 'true',
-			'inline_css'   => true,
-			'timewarp'     => $campaign->timewarp,
-			'ecomm360'     => $campaign->track_orders,
-		);
-
-		if ($this->app->config->deliverance->automatic_analytics_tagging) {
-			$key = $campaign->getAnalyticsKey();
-			if ($key != '') {
-				$options['analytics'] = array('google' => $key);
-			}
-		}
-
-		if ($this->app->config->mail_chimp->default_folder != null) {
-			$options['folder_id'] =
-				$this->app->config->mail_chimp->default_folder;
-		}
-
-		return $options;
-	}
-
-	// }}}
-	// {{{ protected function getCampaignSegmentOptions()
-
-	protected function getCampaignSegmentOptions(
-		DeliveranceMailChimpCampaign $campaign
-	) {
-		$segment_options = $campaign->getSegmentOptions();
-
-		if ($segment_options != null) {
-			if ($this->getSegmentSize($segment_options) == 0) {
-				throw new DeliveranceCampaignException(
-					'Campaign Segment Options return no members');
 			}
-		}
-
-		return $segment_options;
-	}
-
-	// }}}
-	// {{{ protected function getCampaignContent()
-
-	protected function getCampaignContent(
-		DeliveranceMailChimpCampaign $campaign
-	) {
-		$content = array(
-			'html' => $campaign->getContent(DeliveranceCampaign::FORMAT_XHTML),
-			'text' => $campaign->getContent(DeliveranceCampaign::FORMAT_TEXT),
-		);
-
-		return $content;
-	}
-
-	// }}}
-	// {{{ protected function getCampaignsChunk()
-
-	protected function getCampaignsChunk(
-		array $filters = array(),
-		$offset = 0,
-		$chunk_size = 0
-	) {
-		if ($chunk_size > 1000) {
-			throw new DeliveranceException(
-				'Campaign chunk size exceeds API limit');
-		}
-
-		$campaigns = array();
-		// add the list id to the set of passed in filters
-		$filters['list_id'] = $this->shortname;
-
-		$parameters = array(
-			$filters,
-			$offset,
-			$chunk_size,
-			);
-
-		try {
-			$campaigns = $this->callClientMethod(
-				'campaigns',
-				$parameters);
-		} catch (Exception $e) {
-			throw new DeliveranceException($e);
-		}
 
-		return $campaigns;
-	}
-
-	// }}}
-
-	// ecomm360 methods
-	// {{{ public function addOrder()
-
-	public function addOrder(array $order_info)
-	{
-		$success = false;
-
-		try {
-			$parameters = array(
-				$order_info,
+			// Server exceptions
+			if ($last_response['headers']['http_code'] >= 500) {
+				$error = json_decode($last_response['body']);
```

Comment:
Add error handling for failure to decode response JSON. It's possible if MailChimp goes down that we'd get a 503 backend fetch failure that is not JSON-encoded.